lib/improver/nowcasting/optical_flow.py

```diff
@@ -321,23 +321,34 @@ def grid_spacing(coord):
         original_datetime, = \
             (cube.coord("time").units).num2date(cube.coord("time").points)
         new_datetime = original_datetime + timestep
-        new_time = (cube.coord("time").units).date2num(new_datetime)
 
-        if np.int64(new_time) == new_time:
-            new_time = np.int64(new_time)
-        else:
-            new_time = np.float64(new_time)
+        new_time = (cube.coord("time").units).date2num(new_datetime)
 
         advected_cube.coord("time").points = new_time
+        advected_cube.coord("time").convert_units(
+            "seconds since 1970-01-01 00:00:00")
+        advected_cube.coord("time").points = (
+            np.around(advected_cube.coord("time").points).astype(np.int64))
```

Comment:
Would it be possible to have a unit test to show what this around functionality is doing (something with a small amount of precision noise and show it converts properly)?